src/Workspaces/Core/Portable/Workspace/Solution/SolutionState_Checksum.cs

```diff
@@ -19,10 +19,19 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal partial class SolutionState
+    internal partial class SolutionCompilationState
     {
         private static readonly ConditionalWeakTable<IReadOnlyList<ProjectId>, IReadOnlyList<ProjectId>> s_projectIdToSortedProjectsMap = new();
 
+        // Checksums for this solution state
+        private readonly AsyncLazy<SolutionStateChecksums> _lazyChecksums;
```

Comment:
this was one of the parts that made trying to decouple SolutionState and SolutionCompilationState hard.  This ASyncLazy wants to capture the SolutionState instance to do its work.

One option would be to keep the types separate, and move this asynclazy up to Solution itself though.  Overall though, i found it much cleaner to just have SolutionCompilatState point at a SolutionState.